torch/csrc/jit/codegen/cuda/scheduler/registry.h

```diff
@@ -8,6 +8,90 @@ namespace jit {
 namespace fuser {
 namespace cuda {
 
+class SegmentedGroup;
+
+//! SchedulerRuntimeInfo is the abstraction introduced in this PR for passing
+//! runtime
+//!  input dependent information to the schedulers and kernel caches.
+//!
+//! Note:
+//!  if any additional info needed,  or maybe just the inputs themselves it
+//!    could just be added to this class, and they will be distributed to the
+//!    segmenter and schedulers.
+//!  It is important that input id encoding should be up to date with any change
+//!   of this class to avoid launching compiled kernels with illegal inputs.
+class TORCH_CUDA_CU_API SchedulerRuntimeInfo {
+ public:
+  // This constant defines the widest vectorized load/store we
+  //  can do on the target. We should be getting this from some
+  //  library calls. We want to use max vectorized load when we can while
+  //  we don't want to make the max conservatively large since it may create too
+  //  many cache entries for different alignments.
+  const size_t max_alignment_size_in_byte = 128;
+
+  //! Create runtime info for given fusion and input. Creating and binding
+  //! evaluator is
+  //!  optional. The evaluator is used to manage intermediate integers in the
+  //!  fusion. We need them for segmenter and schedulers, but we don't need them
+  //!  when we are just using this class to provide additional encoding for
+  //!  kernel cache lookup.
+  SchedulerRuntimeInfo(
+      Fusion* complete_fusion,
+      const at::ArrayRef<IValue>& inputs,
+      bool create_expr_evaluator = false);
+
+  //! Create runtime info by copying all the global input meta data (i.e.
+  //! alignment), but not
+  //!  the expression evaluator.
+  SchedulerRuntimeInfo(const SchedulerRuntimeInfo& global_runtime_info);
+
+  //! Segmenter facing interface for copying integer values on the segmented
+  //! group inputs
+  //!  so that the intermediates can be computed within the boundary of the
+  //!  segmented group
+  void bindGroupInputs(
+      SegmentedGroup* group,
+      SchedulerRuntimeInfo& global_runtime_info);
+
+  //! Lookup for the alignment sizes of the given tv. Currently only different
+  //! for full fusion
+  //!  tensors and returns a default value for others
+  size_t getAlignmentSize(TensorView* tv);
+
+  //! Take the minimum of input tv alignment sizes. This is both information for
+  //! vectorization and
+  //!  a signature for kernel cache id lookup. May need to be updated with
+  //!  vectorization logic.
+  size_t getCommonAlignmentSize() const {
```

Comment:
This is probably actually not helpful, as we don't care about the alignment of tensors if they're broadcasted on the inner dimension.

Reply:
Sorry, saw how this is used for caching after I put this comment.